explorer/explorerroutes.go

```diff
@@ -1238,6 +1202,23 @@ func (exp *explorerUI) TxPage(w http.ResponseWriter, r *http.Request) {
 		}
 	} // tx.IsTicket()
 
+	// Find atomic swaps related to this tx, if it is a regular, non-coinbase tx.
+	swapsInfo := new(apitypes.TxAtomicSwaps)
+	if tx.Type == txhelpers.TxTypeRegular && !tx.Coinbase {
+		// use following spendingTxsFn to return the already fetched data for
+		// spent outputs instead of re-hitting the database.
+		spendingTxsFn := func(txid string) ([]string, []uint32, []uint32, error) {
+			if txid != tx.TxID {
+				return nil, nil, nil, fmt.Errorf("unexpected txid")
+			}
+			return spendingTxHashes, spendingTxVinInds, voutInds, nil
+		}
+		swapsInfo, err = rpcutils.TxAtomicSwapsInfo(exp.nodeClient, exp.ChainParams, tx.TxID, spendingTxsFn)
```

Comment:
An alternative to passing a `spendingTxsFn` is to pass the already fetched spending info with the call to `rpcutils.TxAtomicSwapsInfo` but that would require modifying the api `getTxSwaps` method to also fetch spending info before calling the rpcutils function. As mentioned earlier, I hope to prevent calls to the db to fetch spending info when there isn't a need to do so.